projects/addon-doc/components/aside/aside.component.ts

```diff
@@ -0,0 +1,57 @@
+import {ChangeDetectionStrategy, Component, Inject, Input} from '@angular/core';
+import {Router} from '@angular/router';
+import {WINDOW} from '@ng-web-apis/common';
+import {TUI_DOC_SCROLL_BEHAVIOR} from '@taiga-ui/addon-doc/tokens';
+
+export interface TuiAsideMenu {
+    id: string;
+    title: string;
+}
+
+@Component({
+    selector: 'tui-doc-aside',
+    templateUrl: './aside.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TuiDocAsideComponent {
+    menuList: readonly TuiAsideMenu[] = [];
+    activeIndex = 0;
+
+    @Input()
+    set menu(menuList: readonly TuiAsideMenu[]) {
+        this.menuList = menuList;
+        this.activeIndex = this.getFragmentIndex();
+    }
+
+    constructor(
+        @Inject(WINDOW) private readonly win: Window,
+        @Inject(Router) private readonly router: Router,
+        @Inject(TUI_DOC_SCROLL_BEHAVIOR) private readonly scrollBehavior: ScrollBehavior,
+    ) {}
+
+    onView(id: string): void {
+        this.win.history.pushState(null, '', `${this.url.base}#${id}`);
```

Comment:
This is done differently when you copy URL of an example. Let's use the same approach in both cases. This one goes behind Angular routing so it's probably better to use the other one.

Reply:
fixed